### PR TITLE
Feature/otp counter

### PR DESCRIPTION
**Pull request by Noiid**

### CODE REVIEW

File: src/controllers/otp.ts [hunk 1]
```diff
@@ -37,22 +37,36 @@ export const request = async (req: Request, res: Response) => {
     }).withMessage('message must contains {code} for unique id')
     .trim()
     .run(req);
-  
+
+  await body('callbackUrl')
```
Comment (taufikdev88) [inline src/controllers/otp.ts]: please use this schema

"callback": {
 "type": ""
 "url": ""
}

File: src/controllers/otp.ts [hunk 2]
```diff
@@ -37,22 +37,36 @@ export const request = async (req: Request, res: Response) => {
...
+    .optional()
+    .matches(/^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$|^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/i).withMessage('invalid format url')
+    .trim()
+    .run(req);
+
+  await body('callbackType')
```
Comment (taufikdev88) [inline src/controllers/otp.ts]: create a enums for type and do a validation here
enum OtpCallbackType {
  Simple = 0
}

File: src/services/otp-service.ts [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 import { Otp } from "../models/otp";
+import { OtpValid } from '../models/otp_validated';
 import logger from "../util/logger";
+import { sendData } from "../util/fetch";
+
+
 
 type GenerationErrors = "INVALID_RECIPIENT_NUMBER" | "ERROR_PERSISTING_OTP";
-type ValidationErrors = "TRANSACTION_NOT_FOUND" | "INVALID_OTP_REF" | "EXPIRED_OTP_TRANSACTION";
+type ValidationErrors = "TRANSACTION_NOT_FOUND" | "INVALID_OTP_REF" | "EXPIRED_OTP_TRANSACTION" | "ERROR_CALLBACK";
```
Comment (taufikdev88) [inline src/services/otp-service.ts]: please change from ERROR _CALLBACK to INVALID_CALLBACK

File: views/otp.pug [hunk 4]
```diff
@@ -62,9 +77,24 @@ append scripts
...
+        $("#callbackUrl").removeAttr("disabled");
+      } else {
+        $("#callbackType").attr("disabled", true);
+        $("#callbackType").val("");
+        $("#callbackUrl").attr("disabled", true);
+        $("#callbackUrl").val("");
+      }
+    });
+
+
```
Comment (taufikdev88) [inline views/otp.pug]: please do not add more enter line

File: src/models/response_callback.ts
Comment (taufikdev88) [inline src/models/response_callback.ts:1]: what is this file used for ?

File: src/models/otp.ts [hunk 5]
```diff
@@ -3,14 +3,20 @@ import mongoose from "mongoose";
 // create an interface representing document in mongoDB
 interface IOtp {
   recipient: string;
+  cs: string;
```
Comment (taufikdev88) [inline src/models/otp.ts]: what is cs ? please use an informatif name

File: src/models/otp_validated.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import mongoose from "mongoose";
+
+// create an interface representing document in mongoDB
+interface IOtpValid {
+    otpNumber: string,
```
Comment (taufikdev88) [inline src/models/otp_validated.ts]: maybe you can change to otpId

**Review by taufikdev88 — COMMENTED** [review]
please kindly check this review first 😬

File: src/models/otp.ts (hunk 5 above)
Comment (Noiid) [reply src/models/otp.ts]: cs is the sender of the OTP validation, is it better to be called the sender?

File: src/models/response_callback.ts
Comment (Noiid) [reply src/models/response_callback.ts:1]: if needed to get data response from api callback

**Review by taufikdev88 — APPROVED** [review]
nice thanks a lot 😄

**Pull request merged**